simplematrixbotlib/bot.py

```diff
@@ -27,11 +27,14 @@ def __init__(self, creds, config = None):
         """
 
         self.creds = creds
+        self.api = botlib.Api(self.creds)
         if config:
             self.config = config
         else:
             self.config = botlib.Config()
-        self.api = botlib.Api(self.creds)
+            # allow (only) users from our own homeserver by default
+            hs: str = self.api.async_client.user_id[self.api.async_client.user_id.index(":")+1:].replace('.', '\\.')
```

Comment:
Wouldn't it be easier to use the `creds.homeserver`?

Reply:
afaik `creds.homeserver` holds the actual server address when the server uses [delegation](https://matrix-org.github.io/synapse/latest/delegate.html).

For example: You can host a simple matrix server and nothing else on "example.org". Then `creds.homeserver == "https://example.org` and matrix IDs are like `@user:example.org`.

You might have a bigger setup and host lots of things on several servers all on your domain and, for easy management of a whole lot of servers, its subdomains, for example website at "example.edu" and email on "mail.example.edu". In this case you can host Matrix at "matrix.example.edu" but configure the server to still use usernames of the base domain. Then `creds.homeserver == "https://matrix.example.edu` (since afaik nio can't resolve delegation?) but matrix IDs are still like `@user:example.org`.

Thus we need to use the latter here, unless nio does that same splitting somewhere, which I however am unaware of.

Ftr my "solution" here to allow the bot's homeserver by default isn't quite optimal so it would probably be great to get your input. Should we really hand a reference to the Bot object to the Config object on construction so it can figure stuff out by itself?